docs/fsharp-cheatsheet.md

```diff
@@ -741,113 +741,165 @@ Another way of implementing interfaces is to use *object expressions*.
+    let readFileTask: Task<string> = readFile "myfile.txt" CancellationToken.None  // (before return) Output: Started Reading Task
+
+    // (readFileTask continues execution on the ThreadPool)
 
-    asyncComputation |> Async.RunSynchronously  // this now runs the async computation
-    newTask.Wait() // this will have already completed by this point
+    let fileContent = readFileTask.Result  // Blocks thread and waits for output. (1)
+    let fileContent' = readFileTask.Result  // Task is already completed, returns same value immediately; no output
 
-    // Output:
-    // Started Task
-    // Created Async
-    // Completed Task
-    // Completed Async
+(1) `.Result` used for demonstration only. Read about [async/await Best Practices](https://learn.microsoft.com/en-us/archive/msdn-magazine/2013/march/async-await-best-practices-in-asynchronous-programming#async-all-the-way)
+
+### Async Computations
+
+Async computations were invented before .NET Tasks existed, which is why F# has two core methods for asynchronous programming. However, async computations did not become obsolete. They offer another, but different, approach: dataflow.
+Async computations are constructed using `async { }` expressions, and the [`Async` module](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-control-fsharpasync.html#section3) is used to compose and execute them.
+In contrast to .NET Tasks, async expressions are "cold" (need to be explicitly started) and every execution [propagates a CancellationToken implicitly](#asynchronous-programming-cancellation-async).
+
+    open System
+    open System.Threading
+    open System.IO
+
+    let readFile filename = async {
+        do! Async.Sleep(TimeSpan.FromSeconds 5)  // use do! when awaiting an Async
+        let! text = File.ReadAllTextAsync(filename) |> Async.AwaitTask  // (1)
+        printfn "Finished Reading File"
+        return text
+    }
 
-### Async and Task Interop
+    // compose a new async computation from exising async computations
+    let readFiles = [ readFile "A"; readFile "B" ] |> Async.Parallel
 
-As F# sits in .NET, and a lot of the codebase uses the C# async/await, the majority of actions are going to be executed and tracked using `System.Threading.Tasks.Task<'T>`s.
+    // execute async computation
+    let textOfFiles: string[] = readFiles |> Async.RunSynchronously
+    // Out: Finished Reading File
+    // Out: Finished Reading File
 
-#### Async.AwaitTask
+    // re-execute async computation again
+    let textOfFiles': string[] = readFiles |> Async.RunSynchronously
+    // Out: Finished Reading File
+    // Out: Finished Reading File
 
-This converts a Task into an async computation. It has the [signature](#functions-signatures): `Task<'T>` -> `Async<'T>`
+(1) As .NET Tasks became the central component of task-based asynchronous programming after F# Async were introduced, F#'s Async has `Async.AwaitTask` for interoperability.
```

Comment:
```suggestion
(1) While `task {}` expressions can invoke an `Async` directly, there's an explicit `Async.AwaitTask` that's used to map from `Task<'T>` to `Async<'T>`. Note there are some [known issues}https://github.com/fsprojects/FSharp.Control.TaskSeq/issues/141) that are [not yet resolved in `FSharp.Core`](https://github.com/fsharp/fslang-suggestions/issues/840), so some codebases use an `AwaitTaskCorrect` helper instead.
```

Reply:
I reworded this a little bit so not to overwhelm the reader. The cancellation section already addresses the issue, and I'll have an exception handling section to mention the other piece.